lib/dates.rb

```diff
@@ -0,0 +1,38 @@
+require 'date'
+require 'pry'
+
+class Dates
+  attr_reader :date, :date_range
+
+  def self.date_format(date)
+    if date.class == Date
+      return date
+    else
```

Comment:
This is more "parsing" (reading a string) than "formatting" (producing a string).